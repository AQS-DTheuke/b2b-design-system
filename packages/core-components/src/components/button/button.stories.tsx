import { Meta, Story } from '@storybook/web-components';
import { html } from 'lit-html';
import buttonDocs from './button.docs.mdx';
import { getArgTypes } from '../../docs/config/utils';

const Template: Story = ({
  variant,
  size,
  disabled,
  loading,
  content,
  type,
  href,
  download,
  target,
}) => {
  const label = content ? content : 'Details';
  return html`<b2b-button
    variant="${variant}"
    size="${size}"
    disabled="${disabled}"
    loading="${loading}"
    type="${type}"
    .href="${href}"
    download="${download}"
    target="${target}">
    ${label}
  </b2b-button>`;
};

const defaultArgs = {
  variant: 'secondary',
  size: '100',
  disabled: false,
  loading: false,
  type: 'button',
  href: undefined,
  target: 'blank',
  download: null,
};

export const story010Secondary = Template.bind({});
story010Secondary.args = { ...defaultArgs };
story010Secondary.storyName = 'Secondary';

export const story020Primary = Template.bind({});
story020Primary.args = {
  ...defaultArgs,
  variant: 'primary',
  content: 'Primary',
};
story020Primary.storyName = 'Primary';

export const story030Loading = Template.bind({});
story030Loading.args = {
  ...defaultArgs,
  variant: 'primary',
  loading: true,
  content: 'Primary',
};
story030Loading.storyName = 'Loading';

export const story040IconStart = Template.bind({});
const iconStartContent = html`<b2b-icon
    slot="start"
    icon="b2b_icon-check"
    size="100"
    color="inherit"></b2b-icon>
  Text`;
story040IconStart.args = {
  ...defaultArgs,
  variant: 'primary',
  content: iconStartContent,
};
story040IconStart.storyName = 'Button With Icon Start';

export const story050IconEnd = Template.bind({});
const iconEndContent = html`Text
  <b2b-icon
    slot="end"
    icon="b2b_icon-check"
    size="100"
    color="inherit"></b2b-icon>`;
story050IconEnd.args = {
  ...defaultArgs,
  variant: 'primary',
  content: iconEndContent,
};
story050IconEnd.storyName = 'Button With Icon End';

export const story055IconStartPali = Template.bind({});
const iconStartPaliContent = html`<i class="obc_icon-check obc_mr-1"></i> Icon
  Button`;
story055IconStartPali.args = {
  ...defaultArgs,
  variant: 'primary',
  content: iconStartPaliContent,
};
story055IconStartPali.storyName = 'Button With Pali Icon';

export const story060IconSearch = Template.bind({});
const iconSearchContent = html`<b2b-icon
  icon="b2b_icon-check"
  size="100"
  color="primary"></b2b-icon>`;
story060IconSearch.args = {
  ...defaultArgs,
  variant: 'secondary',
  content: iconSearchContent,
};
story060IconSearch.storyName = 'Icon Only Button';

export const story070Disabled = Template.bind({});
story070Disabled.args = {
  ...defaultArgs,
  variant: 'primary',
  disabled: true,
  content: 'Primary has grey color when disabled',
};
story070Disabled.storyName = 'Disabled';

export const story080Anchor = Template.bind({});
story080Anchor.args = {
  ...defaultArgs,
  href: 'https://www.otto.de',
  content: 'Anchor Button',
};
story080Anchor.storyName = 'Anchor Button';

export const story090DisabledAnchor = Template.bind({});
story090DisabledAnchor.args = {
  ...defaultArgs,
  href: 'https://www.otto.de',
  disabled: true,
  content: 'Disabled Anchor Button',
};
story090DisabledAnchor.storyName = 'Disabled Anchor Button';

const controls = {
  variant: 'radio',
};
const buttonArgs = getArgTypes('b2b-button', controls);

export default {
  title: 'Components/Interaction/Button',
  argTypes: {
    ...buttonArgs,
    content: {
      control: false,
    },
  },
  viewMode: 'docs',
  parameters: {
    docs: {
      page: buttonDocs,
    },
  },
} as Meta;
