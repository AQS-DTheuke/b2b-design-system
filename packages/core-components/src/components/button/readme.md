# b2b-button



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                                                                                                                                                                           | Type                                           | Default         |
| ---------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------- | --------------- |
| `active`   | `active`   | Whether the button is currently active. Set to true when using the button to signify a permanent active state                                                                         | `boolean`                                      | `false`         |
| `disabled` | `disabled` | Whether the button is disabled.                                                                                                                                                       | `boolean`                                      | `false`         |
| `download` | `download` | If set to true, the browser will attempt to donwload and save the URL instead of opening it. The name of the created file defaults to the URL string, but can be changed by the user. | `string`                                       | `undefined`     |
| `href`     | `href`     | An optional anchor. If specified, the button will render an anchor element that can be use for navigation or download files                                                           | `string`                                       | `undefined`     |
| `loading`  | `loading`  | Whether the button is currently loading. Set to true to hide text and display loading spinner instead                                                                                 | `boolean`                                      | `false`         |
| `size`     | `size`     | The size of the button, 100 is the default state                                                                                                                                      | `"100" \| "50"`                                | `'100'`         |
| `target`   | `target`   | The target of the button if it should behave like an anchor. Per default it is self.                                                                                                  | `"blank" \| "self"`                            | `'self'`        |
| `type`     | `type`     | The button type. Matches native HTML types. The default type is 'button'.                                                                                                             | `"button" \| "reset" \| "submit"`              | `'button'`      |
| `variant`  | `variant`  | The button variant. If not specified, the button will be the secondary variant.                                                                                                       | `"primary" \| "secondary"`                     | `'secondary'`   |
| `width`    | `width`    | The width of the button. Per default, it will fit the content                                                                                                                         | `"custom" \| "fit-container" \| "fit-content"` | `'fit-content'` |


## Methods

### `setFocus() => Promise<void>`

Manually set focus to button element

#### Returns

Type: `Promise<void>`




## Dependencies

### Used by

 - [b2b-calendar-header](../calendar)
 - [b2b-pagination](../pagination)
 - [b2b-search](../search)

### Depends on

- [b2b-spinner](../spinner)

### Graph
```mermaid
graph TD;
  b2b-button --> b2b-spinner
  b2b-calendar-header --> b2b-button
  b2b-pagination --> b2b-button
  b2b-search --> b2b-button
  style b2b-button fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
