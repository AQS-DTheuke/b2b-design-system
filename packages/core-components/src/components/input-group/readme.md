# b2b-input-wrapper



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                                                                                                             | Type      | Default     |
| ---------- | ---------- | ----------------------------------------------------------------------------------------------------------------------- | --------- | ----------- |
| `disabled` | `disabled` | Whether or not the input group is disabled as a whole. Per default it is false.                                         | `boolean` | `false`     |
| `error`    | `error`    | A group error text. It will only show if invalid is set to true.                                                        | `string`  | `undefined` |
| `hint`     | `hint`     | The hint text that appears underneath the input group.                                                                  | `string`  | `undefined` |
| `invalid`  | `invalid`  | Whether or not the input group is invalid and should be rendered with error styles as a whole. Per default it is false. | `boolean` | `false`     |


## Dependencies

### Used by

 - [b2b-search](../search)

### Graph
```mermaid
graph TD;
  b2b-search --> b2b-input-group
  style b2b-input-group fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
