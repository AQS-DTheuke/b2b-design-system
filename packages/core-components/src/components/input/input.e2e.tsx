import { newE2EPage } from '@stencil/core/testing';

describe('B2B-Input', () => {
  let page;

  beforeEach(async () => {
    page = await newE2EPage();
    await page.setContent(`
      <b2b-input></b2b-input>
    `);
  });

  it('should clear input when cleared called', async () => {
    const inputElement = await page.find('b2b-input');

    const nativeInput = await page.find('b2b-input >>> input');
    await nativeInput.type('a value');

    let inputValue = inputElement.getAttribute('value');
    expect(inputValue).toBe('a value');

    await inputElement.callMethod('clearInput');
    await page.waitForChanges();

    inputValue = inputElement.getAttribute('value');
    expect(inputValue).toBe('');
  });

  it('should emit an empty value when cleared', async () => {
    const onInputEventSpy = await page.spyOnEvent('b2b-input');

    const inputElement = await page.find('b2b-input');
    await inputElement.callMethod('clearInput');

    expect(onInputEventSpy).toHaveReceivedEventDetail({ value: '' });
  });
});
