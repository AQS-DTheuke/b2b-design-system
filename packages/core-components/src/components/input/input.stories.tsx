import { Meta, Story } from '@storybook/web-components';
import { html } from 'lit-html';
import inputDocs from './input.docs.mdx';
import { getArgTypes } from '../../docs/config/utils';

const Template: Story = ({
  label,
  required,
  type,
  disabled,
  placeholder,
  value,
  hint,
  error,
  autofocus,
  invalid,
  iconMarkup,
}) => {
  const defaultLabel = label ? label : 'Input Label';
  const defaultType = type ? type : 'text';
  return html`<div style="width: 400px;">
    <b2b-input
      label="${defaultLabel}"
      required="${required}"
      type="${defaultType}"
      disabled="${disabled}"
      placeholder="${placeholder}"
      value="${value}"
      hint="${hint}"
      error="${error}"
      autofocus="${autofocus}"
      invalid=${invalid}
      >${iconMarkup}</b2b-input
    >
  </div>`;
};

const defaultArgs = {
  label: 'Label',
  required: false,
  type: 'text',
  disabled: false,
  placeholder: '',
  value: '',
  hint: '',
  error: '',
  autofocus: false,
  invalid: false,
};

export const story010Default = Template.bind({});
story010Default.args = { ...defaultArgs, label: 'Default Input' };
story010Default.storyName = 'Default';

export const story020Placeholder = Template.bind({});
story020Placeholder.args = {
  ...defaultArgs,
  label: 'Input with Placeholder',
  placeholder: 'This is a placeholder',
};
story020Placeholder.storyName = 'Placeholder';

export const story030Disabled = Template.bind({});
story030Disabled.args = {
  ...defaultArgs,
  label: 'Disabled Input',
  disabled: true,
};
story030Disabled.storyName = 'Disabled';

export const story040Hint = Template.bind({});
story040Hint.args = {
  ...defaultArgs,
  label: 'Input with hint',
  hint: 'This is a hint',
};
story040Hint.storyName = 'Hint';

export const story050Error = Template.bind({});
story050Error.args = {
  ...defaultArgs,
  label: 'Input with an error',
  error: 'This is an error',
  invalid: true,
};
story050Error.storyName = 'Error';

export const story055Required = Template.bind({});
story055Required.args = {
  ...defaultArgs,
  label: 'Required input',
  required: true,
};
story055Required.storyName = 'Required';

export const story060IconEnd = Template.bind({});
const icon = html`<b2b-icon icon="b2b_icon-search" slot="end"></b2b-icon>`;
story060IconEnd.args = { ...defaultArgs, iconMarkup: icon };
story060IconEnd.storyName = 'Input with Icon';

export const story080Prefix = Template.bind({});
const prefix = html`<span slot="start">Registration Number</span>`;
story080Prefix.args = { ...defaultArgs, iconMarkup: prefix };
story080Prefix.storyName = 'Input with Prefix';

export const story090PrefixSuffix = Template.bind({});
const suffix = html`<span slot="start">Registration Number</span
  ><b2b-icon icon="b2b_icon-search" slot="end"></b2b-icon>`;
story090PrefixSuffix.args = { ...defaultArgs, iconMarkup: suffix };
story090PrefixSuffix.storyName = 'Input with Prefix and Icon';

export const story070AllArgs = Template.bind({});
story070AllArgs.args = {
  ...defaultArgs,
  label: 'Input with all arguments',
  required: true,
  placeholder: 'All arguments placeholder',
  value: 'an initial value',
  hint: 'This is a hint for an every argument specified input',
  error: 'This is an error so you will not see the hint',
  disabled: false,
  autofocus: false,
};
story070AllArgs.storyName = 'All arguments specified';

const controls = {
  type: 'radio',
};

const inputArgs = getArgTypes('b2b-input', controls);

export default {
  title: 'Components/Form/Input',
  argTypes: {
    ...inputArgs,
    iconMarkup: {
      control: false,
    },
  },
  viewMode: 'docs',
  parameters: {
    docs: {
      page: inputDocs,
    },
  },
} as Meta;
