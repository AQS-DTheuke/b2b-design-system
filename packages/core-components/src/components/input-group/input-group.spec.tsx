import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { B2BInputGroup } from './input-group';
import { InputComponent } from '../input/input';
import { ButtonComponent } from '../button/button';
import { DropdownComponent } from '../dropdown/dropdown';

describe('B2B Input Group', () => {
  it('should render content wrapped in the input group', async () => {
    const page = await newSpecPage({
      components: [
        B2BInputGroup,
        InputComponent,
        ButtonComponent,
        DropdownComponent,
      ],
      template: () => (
        <b2b-input-group>
          <b2b-input label="Search Term"></b2b-input>
          <b2b-dropdown>
            <option value="data-1">Data Set 1</option>
            <option value="data-2">Data Set 2</option>
          </b2b-dropdown>
          <b2b-button variant="primary">Search</b2b-button>
        </b2b-input-group>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should render a group hint if it is specified', async () => {
    const page = await newSpecPage({
      components: [
        B2BInputGroup,
        InputComponent,
        ButtonComponent,
        DropdownComponent,
      ],
      template: () => (
        <b2b-input-group hint="A group hint">
          <b2b-input label="Search Term"></b2b-input>
          <b2b-dropdown>
            <option value="data-1">Data Set 1</option>
            <option value="data-2">Data Set 2</option>
          </b2b-dropdown>
          <b2b-button variant="primary">Search</b2b-button>
        </b2b-input-group>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should render a group error if one is specified and the group is invalid', async () => {
    const page = await newSpecPage({
      components: [
        B2BInputGroup,
        InputComponent,
        ButtonComponent,
        DropdownComponent,
      ],
      template: () => (
        <b2b-input-group error="A group error" invalid>
          <b2b-input label="Search Term"></b2b-input>
          <b2b-dropdown>
            <option value="data-1">Data Set 1</option>
            <option value="data-2">Data Set 2</option>
          </b2b-dropdown>
          <b2b-button variant="primary">Search</b2b-button>
        </b2b-input-group>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should not render a group error if the group is invalid and disabled', async () => {
    const page = await newSpecPage({
      components: [
        B2BInputGroup,
        InputComponent,
        ButtonComponent,
        DropdownComponent,
      ],
      template: () => (
        <b2b-input-group
          invalid
          disabled
          hint="A group hint"
          error="A group error">
          <b2b-input label="Search Term"></b2b-input>
          <b2b-dropdown>
            <option value="data-1">Data Set 1</option>
            <option value="data-2">Data Set 2</option>
          </b2b-dropdown>
          <b2b-button variant="primary">Search</b2b-button>
        </b2b-input-group>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
