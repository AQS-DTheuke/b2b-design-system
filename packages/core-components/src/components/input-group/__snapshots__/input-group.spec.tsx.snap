// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`B2B Input Group should not render a group error if the group is invalid and disabled 1`] = `
<b2b-input-group>
  <mock:shadow-root>
    <div class="b2b-input-wrapper">
      <slot name="start"></slot>
      <slot></slot>
      <slot name="end"></slot>
    </div>
    <span class="b2b-input-wrapper-hint">
      A group hint
    </span>
  </mock:shadow-root>
  <b2b-input class="b2b-input" disabled="" type="text">
    <mock:shadow-root>
      <b2b-input-label>
        Search Term
      </b2b-input-label>
      <div class="input-wrapper input-wrapper--disabled">
        <slot name="start"></slot>
        <input class="native-input" disabled="" type="text">
        <slot name="end"></slot>
      </div>
    </mock:shadow-root>
  </b2b-input>
  <b2b-dropdown class="b2b-dropdown b2b-dropdown--disabled" disabled="">
    <mock:shadow-root>
      <select class="b2b-dropdown-select" disabled="">
        <option value="data-1">
          Data Set 1
        </option>
        <option value="data-2">
          Data Set 2
        </option>
      </select>
    </mock:shadow-root>
  </b2b-dropdown>
  <b2b-button class="button-100" disabled="">
    <mock:shadow-root>
      <button class="primary" disabled="" type="button">
        <span>
          <slot name="start"></slot>
          <slot></slot>
          <slot name="end"></slot>
        </span>
      </button>
    </mock:shadow-root>
    Search
  </b2b-button>
</b2b-input-group>
`;

exports[`B2B Input Group should render a group error if one is specified and the group is invalid 1`] = `
<b2b-input-group>
  <mock:shadow-root>
    <div class="b2b-input-wrapper">
      <slot name="start"></slot>
      <slot></slot>
      <slot name="end"></slot>
    </div>
    <span class="b2b-input-wrapper-hint b2b-input-wrapper-hint--error">
      A group error
    </span>
  </mock:shadow-root>
  <b2b-input class="b2b-input b2b-input--error" invalid="" type="text">
    <mock:shadow-root>
      <b2b-input-label>
        Search Term
      </b2b-input-label>
      <div class="input-wrapper">
        <slot name="start"></slot>
        <input class="native-input" type="text">
        <slot name="end"></slot>
      </div>
    </mock:shadow-root>
  </b2b-input>
  <b2b-dropdown class="b2b-dropdown b2b-dropdown--error" invalid="">
    <mock:shadow-root>
      <select class="b2b-dropdown-select">
        <option value="data-1">
          Data Set 1
        </option>
        <option value="data-2">
          Data Set 2
        </option>
      </select>
    </mock:shadow-root>
  </b2b-dropdown>
  <b2b-button class="button-100">
    <mock:shadow-root>
      <button class="primary" type="button">
        <span>
          <slot name="start"></slot>
          <slot></slot>
          <slot name="end"></slot>
        </span>
      </button>
    </mock:shadow-root>
    Search
  </b2b-button>
</b2b-input-group>
`;

exports[`B2B Input Group should render a group hint if it is specified 1`] = `
<b2b-input-group>
  <mock:shadow-root>
    <div class="b2b-input-wrapper">
      <slot name="start"></slot>
      <slot></slot>
      <slot name="end"></slot>
    </div>
    <span class="b2b-input-wrapper-hint">
      A group hint
    </span>
  </mock:shadow-root>
  <b2b-input class="b2b-input" type="text">
    <mock:shadow-root>
      <b2b-input-label>
        Search Term
      </b2b-input-label>
      <div class="input-wrapper">
        <slot name="start"></slot>
        <input class="native-input" type="text">
        <slot name="end"></slot>
      </div>
    </mock:shadow-root>
  </b2b-input>
  <b2b-dropdown class="b2b-dropdown">
    <mock:shadow-root>
      <select class="b2b-dropdown-select">
        <option value="data-1">
          Data Set 1
        </option>
        <option value="data-2">
          Data Set 2
        </option>
      </select>
    </mock:shadow-root>
  </b2b-dropdown>
  <b2b-button class="button-100">
    <mock:shadow-root>
      <button class="primary" type="button">
        <span>
          <slot name="start"></slot>
          <slot></slot>
          <slot name="end"></slot>
        </span>
      </button>
    </mock:shadow-root>
    Search
  </b2b-button>
</b2b-input-group>
`;

exports[`B2B Input Group should render content wrapped in the input group 1`] = `
<b2b-input-group>
  <mock:shadow-root>
    <div class="b2b-input-wrapper">
      <slot name="start"></slot>
      <slot></slot>
      <slot name="end"></slot>
    </div>
  </mock:shadow-root>
  <b2b-input class="b2b-input" type="text">
    <mock:shadow-root>
      <b2b-input-label>
        Search Term
      </b2b-input-label>
      <div class="input-wrapper">
        <slot name="start"></slot>
        <input class="native-input" type="text">
        <slot name="end"></slot>
      </div>
    </mock:shadow-root>
  </b2b-input>
  <b2b-dropdown class="b2b-dropdown">
    <mock:shadow-root>
      <select class="b2b-dropdown-select">
        <option value="data-1">
          Data Set 1
        </option>
        <option value="data-2">
          Data Set 2
        </option>
      </select>
    </mock:shadow-root>
  </b2b-dropdown>
  <b2b-button class="button-100">
    <mock:shadow-root>
      <button class="primary" type="button">
        <span>
          <slot name="start"></slot>
          <slot></slot>
          <slot name="end"></slot>
        </span>
      </button>
    </mock:shadow-root>
    Search
  </b2b-button>
</b2b-input-group>
`;
